frontend/src/components/mine/Reports/MineReportInfo.js

```diff
@@ -27,46 +34,98 @@ const propTypes = {
   mines: PropTypes.objectOf(CustomPropTypes.mine).isRequired,
   mineGuid: PropTypes.string.isRequired,
   mineReports: PropTypes.arrayOf(CustomPropTypes.mineReport).isRequired,
+  mineReportDefinitionOptions: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
   fetchMineReports: PropTypes.func.isRequired,
   updateMineReport: PropTypes.func.isRequired,
   createMineReport: PropTypes.func.isRequired,
   deleteMineReport: PropTypes.func.isRequired,
+  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
   openModal: PropTypes.func.isRequired,
   closeModal: PropTypes.func.isRequired,
+  location: PropTypes.shape({ search: PropTypes.string }).isRequired,
+  match: PropTypes.shape({
+    params: {
+      id: PropTypes.string,
+    },
+  }).isRequired,
+};
+
+const initialSearchValues = {
+  report_name: "",
+  report_type: "",
+  compliance_year: "",
+  report_due_date_start: "",
+  report_due_date_end: "",
+  report_status: "",
 };
 
 export class MineReportInfo extends Component {
   state = {
     mine: {},
+    reportFilterParams: initialSearchValues,
+    filteredReports: [],
   };
 
   componentWillMount = () => {
-    this.props.fetchMineReports(this.props.mineGuid);
+    this.props.fetchMineReports(this.props.mineGuid).then(() => {
+      if (Object.keys(this.props.location.search).length > 0) {
+        this.renderDataFromURL(this.props.location.search);
```

Comment:
`this.props.location.search` is a string, checking its existance is enough
```
if (this.props.location.search) {
// do something
}
```

Reply:
When I was out testing it wasn't a string when it was empty. It was an empty dictionary.

Reply:
we've been using `queryString.parse(this.props.location.search)` to turn 
`page=5&per_page=25`
 into 
`{
page: 5,
per_page: 25
}`

are you sure you were logging the correct thing?

Reply:
You are correct, I dont know what was going on yesterday but every time I output this is was an empty object and that code worked... I have changed it but super confused.